server/src/storage/data_transfer.rs

```diff
@@ -0,0 +1,399 @@
+/* Copyright 2022 The ModelarDB Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//! Support for efficiently transferring data to a remote blob store. Data saved locally on disk is
+//! managed here until it is of a sufficient size to be transferred efficiently.
+
+use std::collections::HashMap;
+use std::io::Error as IOError;
+use std::io::ErrorKind::Other;
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+
+use bytes::BufMut;
+use datafusion::arrow::compute;
+use datafusion::arrow::record_batch::RecordBatch;
+use datafusion::parquet::errors::ParquetError;
+use object_store::path::{Path as ObjectStorePath, PathPart};
+use object_store::ObjectStore;
+use object_store::local::LocalFileSystem;
+use tokio::runtime::Runtime;
+use futures::StreamExt;
+use tonic::codegen::Bytes;
+use tracing::{debug, debug_span};
+
+use crate::{storage, StorageEngine};
+
+// TODO: When the storage engine is changed to use object store for everything, receive
+//       the object store directly through the parameters instead.
+// TODO: Handle the case where a connection can not be established when transferring data.
+// TODO: If there is a remote data folder, initialize the data transfer component in main.
+// TODO: When compressed data is saved, add the compressed file to the data transfer component.
+
+pub struct DataTransfer {
+    /// Tokio runtime for executing asynchronous tasks.
+    runtime: Arc<Runtime>,
+    /// Path to the folder containing all compressed data managed by the [`StorageEngine`].
+    data_folder_path: PathBuf,
```

Comment:
Maybe use `local_data_folder_path` for consistency with `DataFolders` in `main.rs`.

Reply:
Fixed.